File: src/main/java/Solution_509_FibonacciNumber.java
public class Solution_509_FibonacciNumber {
    public static void main(String[] args) {
        System.out.println(fibonacci(8));
    }

    public static int fibonacci(Integer n) {
        if (n == 0 || n == 1) return n;
        return fibonacci(n - 1) + fibonacci(n - 2);
    }
}

File: src/main/kotlin/Solution_509_FibonacciNumber.kt
/**
 * 509 Fibonacci Number
 *
 * The Fibonacci numbers, commonly denoted F(n) form a sequence, called the Fibonacci sequence, such that each number is the sum of the two preceding ones, starting from 0 and 1. That is,
 *
 * F(0) = 0, F(1) = 1
 *
 * F(n) = F(n - 1) + F(n - 2), for n > 1.
 *
 * Given n, calculate F(n).
 */
fun main() {
    val n = 8
    println(fibonacci(n))
}

fun fibonacci(n: Int): Int {
    if (n == 0 || n == 1) return n
    return fibonacci(n - 1) + fibonacci(n - 2)
}